### PR TITLE
ML4PhotoConv: more photon variables

### PR DESCRIPTION
**Pull request by an15ms157**
More variables added for PCM Machine Learning

### CODE REVIEW

File: PWGGA/GammaConv/AliAnalysisTaskGammaConvV1.cxx [hunk 1]
```diff
@@ -3780,33 +3930,33 @@ void AliAnalysisTaskGammaConvV1::CalculatePi0Candidates(){
...
-        gamma0->GetTrackLabelNegative() == gamma1->GetTrackLabelPositive() ||
-        gamma0->GetTrackLabelPositive() == gamma1->GetTrackLabelNegative() ) continue;
-
-        AliAODConversionMother *pi0cand = new AliAODConversionMother(gamma0,gamma1);
-        pi0cand->SetLabels(firstGammaIndex,secondGammaIndex);
-        pi0cand->CalculateDistanceOfClossetApproachToPrimVtx(fInputEvent->GetPrimaryVertex());
-        
-        AliVTrack * ENegTrack_Gamma0 = fiPhotonCut->GetTrack(fInputEvent, gamma0->GetTrackLabelNegative());
-        AliVTrack * EPosTrack_Gamma0 = fiPhotonCut->GetTrack(fInputEvent, gamma0->GetTrackLabelPositive());
+    if (fDoTreeForMesonML){
```
Comment (jokonig) [inline PWGGA/GammaConv/AliAnalysisTaskGammaConvV1.cxx:3933]: Have you checked if the normal analysis is still working? The if(fDoTreeForMesonML) makes me wonder if the pi0 calculation is now only enabled for the ML part

File: PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_ConvMode_pPb.C [hunk 2]
```diff
@@ -444,7 +444,7 @@ void AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_ConvMode_pPb(
     }
     if( ! analysisCuts[i]->InitializeCutsFromCutString((cuts.GetPhotonCut(i)).Data()) ) {
       cout<<"ERROR: analysisCuts [" <<i<<"]"<<endl;
-      return;
+      return 0;
```
Comment (jokonig) [inline PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_ConvMode_pPb.C:447]: ?? Why add the 0 there? I think the function is void if it is the AddTask

File: PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_MixedMode_pPb.C [hunk 3]
```diff
@@ -584,7 +584,7 @@ AliVEventHandler *inputHandler=mgr->GetInputEventHandler();
     if(runLightOutput>0) analysisCuts[i]->SetLightOutput(kTRUE);
     if( ! analysisCuts[i]->InitializeCutsFromCutString((cuts.GetPhotonCut(i)).Data()) ) {
       cout<<"ERROR: analysisCuts [" <<i<<"]"<<endl;
-      return;
+      return 0;
```
Comment (jokonig) [inline PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_MixedMode_pPb.C:587]: Why add 0 here??

File: PWGGA/GammaConv/AliAnalysisTaskGammaConvV1.h [hunk 4]
```diff
@@ -388,6 +389,12 @@ class AliAnalysisTaskGammaConvV1 : public AliAnalysisTaskSE {
     Float_t                           fEta_PhotonML;                              
     Float_t                           fPhi_PhotonML;                              
     Float_t                           fR_PhotonML;                              
+    Float_t                           fCosThetaPA_PhotonML;
```
Comment (jokonig) [inline PWGGA/GammaConv/AliAnalysisTaskGammaConvV1.h:392]: In principle we dont need to stream all your variables. You could indicate that they should not be streamed be commenting //! (see https://alice-doc.github.io/alice-analysis-tutorial/analysis/streamer.html)

**Review by jokonig — CHANGES_REQUESTED** [review]

File: PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_ConvMode_pPb.C (hunk 2 above)
Comment (an15ms157) [reply PWGGA/GammaConv/macros/AddTask_GammaConvNeutralMesonPiPlPiMiNeutralMeson_ConvMode_pPb.C:447]: Yes, this does not seem right. Probably a typo, I'll change them

**Comment by jokonig** [comment]
+1

**Pull request merged**